src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -786,4 +786,41 @@
       it { is_expected.to redirect_to(action: :show, project: user.home_project) }
     end
   end
+
+  describe "POST #add_maintained_project" do
+    before do
+      login user
+    end
+
+    context "maintained project successfully added" do
+      before do
+        user.home_project.update(kind: 'maintenance')
+      end
+
+      context "maintained project successfully added" do
+        before do
+          post :add_maintained_project, project: user.home_project, maintained_project: user.home_project.name
+        end
+
+        it { expect(flash[:notice]).to eq("Added #{user.home_project.name} to maintenance") }
+        it { is_expected.to redirect_to(action: 'maintained_projects', project: user.home_project) }
+      end
+
+      # raise the exception in the before_filter set_maintained_project
+      it "#add_maintained_project raise excepction with invalid maintained project" do
+        expect {
+          post :add_maintained_project, project: user.home_project, maintained_project: "invalid"
+        }.to raise_exception ActiveRecord::RecordNotFound
+      end
+    end
+
+    # redirect in the before_filter require_maintenance_project
+    context "#add_maintained_project fails without maintenance kind for a valid maintained project" do
```

Comment:
I wouldn't add a comment for every before_filter. Just for places where it might not be that obvious what happens. Like the case with the raised ActiveRecord::RecordNotFound exception.
When I saw that in the test I thought this might mean that we do not catch this at all. There a comment is very helpful.